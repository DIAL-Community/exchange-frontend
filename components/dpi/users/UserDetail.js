import { useCallback } from 'react'
import { FormattedDate, FormattedTime, useIntl } from 'react-intl'
import { useUser } from '../../../lib/hooks'
import EditButton from '../../shared/form/EditButton'

const UserDetail = ({ user }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { user: loggedInUser } = useUser()

  const editPath = `${user.id}/edit`

  return (
    <div className='relative flex flex-col gap-y-3'>
      {(loggedInUser.isAdliAdminUser || loggedInUser.isAdminUser) && (
        <div className='absolute top-1 -right-1'>
          <EditButton type='link' href={editPath} />
        </div>
      )}
      <div className='text-xl font-semibold py-3'>
        {format('ui.common.detail.description')}
      </div>
      <div className='block'>
        {user.email}
      </div>
      <hr className='border-b border-dial-blue-chalk my-3' />
      <div className='flex flex-col gap-y-3'>
        <div className='text-xl font-semibold pb-3'>
          {format('user.roles')}
        </div>
        <div className='text-sm'>
          {user?.roles.map(x => x.toUpperCase()).join(', ')}
        </div>
      </div>
      <hr className='border-b border-dial-blue-chalk my-3' />
      <div className='flex flex-col gap-y-3'>
        <div className='font-semibold'>
          {format('ui.user.accountDetails')}
        </div>
        <div className='text-xs italic'>
          {format('ui.user.createdAt')}:&nbsp;
          <FormattedDate value={user.createdAt} />&nbsp;
          <FormattedTime value={user.createdAt} />
        </div>
        {`${user.confirmed}` === 'true' &&
          <div className='text-xs italic'>
            {format('ui.user.confirmedAt')}:&nbsp;
            <FormattedDate value={user.confirmedAt} />&nbsp;
            <FormattedTime value={user.confirmedAt} />
          </div>
        }
      </div>
    </div>
  )
}

export default UserDetail
