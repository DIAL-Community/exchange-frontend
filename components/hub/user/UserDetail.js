import { useCallback } from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { FiEdit3 } from 'react-icons/fi'
import { FormattedDate, FormattedTime, useIntl } from 'react-intl'
import { useUser } from '../../../lib/hooks'

const UserDetail = ({ user }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { user: loggedInUser } = useUser()

  const { asPath } = useRouter()
  const generateEditPath = () => {
    if (asPath.indexOf('/hub/admin/profile') >= 0) {
      return '/hub/admin/profile/edit-user'
    } else {
      return `/hub/admin/users/${user.id}/edit-user`
    }
  }

  return (
    <div className='relative flex flex-col gap-y-3'>
      {(loggedInUser.isAdliAdminUser || loggedInUser.isAdminUser) && (
        <div className='cursor-pointer absolute -top-2 -right-1'>
          <Link href={generateEditPath()} className='bg-dial-iris-blue px-3 py-2 rounded text-white'>
            <FiEdit3 className='inline pb-0.5' />
            <span className='text-sm px-1'>
              {`${format('app.edit')} ${format('ui.user.label')}`}
            </span>
          </Link>
        </div>
      )}
      <div className='flex flex-col gap-y-3'>
        <div className='font-semibold'>
          {format('user.email')}
        </div>
        <div className='description-block'>
          {user.email ?? user.userEmail}
        </div>
      </div>
      <hr className='border-b border-dial-blue-chalk my-3' />
      <div className='flex flex-col gap-y-3'>
        <div className='font-semibold'>
          {format('user.roles')}
        </div>
        <div className='text-sm'>
          {user?.roles.map(x => x.toUpperCase()).join(', ')}
        </div>
      </div>
      <hr className='border-b border-dial-blue-chalk my-3' />
      <div className='flex flex-col gap-y-3'>
        <div className='font-semibold'>
          {format('ui.user.accountDetails')}
        </div>
        <div className='text-xs italic'>
          {format('ui.user.createdAt')}:&nbsp;
          <FormattedDate value={user.createdAt} />&nbsp;
          <FormattedTime value={user.createdAt} />
        </div>
        {`${user.confirmed}` === 'true' &&
          <div className='text-xs italic'>
            {format('ui.user.confirmedAt')}:&nbsp;
            <FormattedDate value={user.confirmedAt} />&nbsp;
            <FormattedTime value={user.confirmedAt} />
          </div>
        }
      </div>
    </div>
  )
}

export default UserDetail
