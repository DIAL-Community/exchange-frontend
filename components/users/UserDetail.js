import { useIntl } from 'react-intl'
import { useSession } from 'next-auth/client'

const UserDetail = ({ user }) => {
  const { formatMessage } = useIntl()
  const format = (id, values) => formatMessage({ id }, values)

  const [session] = useSession()

  return (
    <>
      <div className='flex flex-col lg:flex-row justify-between pb-8 max-w-catalog mx-auto'>
        <div className='relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4'>
          {
            session && session.user.canEdit && (
              <a href={`/users/${user.id}/edit`} className='bg-dial-blue px-2 py-1 rounded text-white mr-5 mb-5'>
                <img src='/icons/edit.svg' className='inline mr-2 pb-1' alt='Edit' height='12px' width='12px' />
                <span className='text-sm px-2'>{format('app.edit')}</span>
              </a>
            )
          }
          <div className='bg-white border-2 border-dial-gray lg:mr-6 shadow-lg'>
            <div className='flex flex-col h-80 p-4'>
              <div className='text-2xl font-semibold absolute w-4/5 md:w-auto lg:w-64 2xl:w-80 bg-white bg-opacity-80 text-dial-purple'>
                {format('profile.username')} : {user.username}
              </div>
              <div className='pt-8 m-auto align-middle w-48'>
                <img
                  alt={format('image.alt.logoFor', { name: user.username })}
                  src={user.image}
                />
              </div>
              <div className='text-sm text-center text-dial-gray-dark'>
                {format('profile.email')} : {user.email}
              </div>
            </div>
          </div>
        </div>
        <div className='w-full lg:w-2/3 xl:w-3/4'>
          <div className='my-4 h2'>
            {format('profile.profile')}{user.username}
          </div>
          <div className='my-3 h4'>
            {format('profile.roles')} {user.roles.map(role => role.toUpperCase() + '; ')}
          </div>
          <div className='my-3 h4'>
            {format('profile.products')} {user.products.map(prod => prod.name)}
          </div>
          <div className='my-3 h4'>
            {format('profile.organization')} {user.organization && user.organization.name}
          </div>
        </div>
      </div>
    </>
  )
}

export default UserDetail
