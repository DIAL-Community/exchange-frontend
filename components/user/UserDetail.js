import { useEffect, useRef, useState } from 'react'
import { useApolloClient, useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../lib/apolloClient'
import Breadcrumb from '../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../shared/GraphQueryHandler'
import { USER_DETAIL_QUERY, USER_POLICY_QUERY } from '../shared/query/user'
import { fetchOperationPolicies } from '../utils/policy'
import UserDetailLeft from './UserDetailLeft'
import UserDetailRight from './UserDetailRight'

const UserDetail = ({ userId }) => {
  const scrollRef = useRef(null)
  const client = useApolloClient()

  const [editingAllowed, setEditingAllowed] = useState(false)
  const [deletingAllowed, setDeletingAllowed] = useState(false)

  const { loading, error, data } = useQuery(USER_DETAIL_QUERY, {
    variables: { userId },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING
      }
    }
  })

  useEffect(() => {
    fetchOperationPolicies(
      client,
      USER_POLICY_QUERY,
      ['editing', 'deleting']
    ).then(policies => {
      setEditingAllowed(policies['editing'])
      setDeletingAllowed(policies['deleting'])
    })
  }, [client])

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.user) {
    return handleMissingData()
  }

  const { user } = data

  const slugNameMapping = (() => {
    const map = {}
    map[user.id] = user.email

    return map
  })()

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-violet text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3 shrink-0'>
          <UserDetailLeft scrollRef={scrollRef} user={user} />
        </div>
        <div className='lg:basis-2/3'>
          <UserDetailRight
            ref={scrollRef}
            user={user}
            editingAllowed={editingAllowed}
            deletingAllowed={deletingAllowed}
          />
        </div>
      </div>
    </div>
  )
}

export default UserDetail
