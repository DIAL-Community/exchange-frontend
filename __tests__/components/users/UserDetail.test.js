import { mockRouterImplementation, mockSessionImplementation, render } from '../../test-utils'
import UserDetail from '../../../components/users/UserDetail'
import { basicUser, user, userWithProducts } from './data/UserDetail'

jest.mock('next/dist/client/router')
jest.mock('next-auth/client')

describe('Unit test for the UserDetail component', () => {
  beforeAll(() => {
    mockRouterImplementation()
    mockSessionImplementation(true)
  })

  test('should not render 0 for user not owning any product.', () => {
    const component = render(<UserDetail user={user} />)

    // Should not display the products section.
    expect(component.queryByText(/0/)).not.toBeInTheDocument()
    expect(component.queryByText(/Products/)).not.toBeInTheDocument()
    // Should display the role
    expect(component.getByText(/ORG_USER/)).toBeInTheDocument()
    // Should display the organization name.
    expect(component.getByText(/Some Organization/)).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })

  test('should render element of the user owning product.', () => {
    const component = render(<UserDetail user={userWithProducts} />)

    // Should correctly format both roles.
    expect(component.getByText(/ORG_USER, PRODUCT_USER/)).toBeInTheDocument()

    // Should display both the organization and product name.
    expect(component.getByText(/Some Organization/)).toBeInTheDocument()
    expect(component.getByText(/Some Product/)).toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })

  test('should render element of basic user not owning product or organization.', () => {
    const component = render(<UserDetail user={basicUser} />)

    expect(component.getByText(/USER/)).toBeInTheDocument()
    // Should not display the products and organizations section.
    expect(component.queryByText(/Products/)).not.toBeInTheDocument()
    expect(component.queryByText(/Organization/)).not.toBeInTheDocument()
    expect(component).toMatchSnapshot()
  })
})
