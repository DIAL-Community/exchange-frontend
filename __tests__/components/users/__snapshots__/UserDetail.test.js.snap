// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for the UserDetail component should not render 0 for user not owning any product. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Toastify"
      />
      <div
        class="flex flex-col lg:flex-row justify-between pb-8"
      >
        <div
          class="relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4"
        >
          <div
            class="pb-4 flex"
          >
            <a
              class="cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
              data-testid="edit-link"
              href="/users/undefined/edit"
            >
              <svg
                class="inline pb-0.5"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 20h9"
                />
                <path
                  d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
                />
              </svg>
              <span
                class="text-sm px-1"
              >
                Edit
              </span>
            </a>
          </div>
          <div
            class="bg-white border-2 border-dial-gray lg:mr-6 shadow-lg"
          >
            <div
              class="flex flex-col h-80 p-4"
            >
              <div
                class="m-auto align-middle w-48"
              >
                <img
                  alt="Logo for: Some User"
                  class="w-24 m-auto"
                  src="/icons/user.svg"
                />
              </div>
              <div
                class="text-sm text-center text-dial-gray-dark"
              >
                Email
                 : 
                some_user@some-website.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full lg:w-2/3 xl:w-3/4 flex flex-col"
        >
          <div
            class="hidden lg:block"
          >
            <div
              class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
            >
              <a
                class="inline text-dial-sapphire h5"
                href="/"
              >
                Home
              </a>
              <div
                class="inline h5"
              >
                 &gt; 
                <a
                  class="text-dial-gray-dark"
                  href="/"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                User Profile
                :
              </div>
              <div>
                Some User
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                Roles
                :
              </div>
              <div>
                ORG_USER
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                Organization
                :
              </div>
              <div>
                Some Organization
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Toastify"
    />
    <div
      class="flex flex-col lg:flex-row justify-between pb-8"
    >
      <div
        class="relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4"
      >
        <div
          class="pb-4 flex"
        >
          <a
            class="cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
            data-testid="edit-link"
            href="/users/undefined/edit"
          >
            <svg
              class="inline pb-0.5"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 20h9"
              />
              <path
                d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
              />
            </svg>
            <span
              class="text-sm px-1"
            >
              Edit
            </span>
          </a>
        </div>
        <div
          class="bg-white border-2 border-dial-gray lg:mr-6 shadow-lg"
        >
          <div
            class="flex flex-col h-80 p-4"
          >
            <div
              class="m-auto align-middle w-48"
            >
              <img
                alt="Logo for: Some User"
                class="w-24 m-auto"
                src="/icons/user.svg"
              />
            </div>
            <div
              class="text-sm text-center text-dial-gray-dark"
            >
              Email
               : 
              some_user@some-website.com
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full lg:w-2/3 xl:w-3/4 flex flex-col"
      >
        <div
          class="hidden lg:block"
        >
          <div
            class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
          >
            <a
              class="inline text-dial-sapphire h5"
              href="/"
            >
              Home
            </a>
            <div
              class="inline h5"
            >
               &gt; 
              <a
                class="text-dial-gray-dark"
                href="/"
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-3"
        >
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              User Profile
              :
            </div>
            <div>
              Some User
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              Roles
              :
            </div>
            <div>
              ORG_USER
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              Organization
              :
            </div>
            <div>
              Some Organization
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Unit test for the UserDetail component should render element of basic user not owning product or organization. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Toastify"
      />
      <div
        class="flex flex-col lg:flex-row justify-between pb-8"
      >
        <div
          class="relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4"
        >
          <div
            class="pb-4 flex"
          >
            <a
              class="cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
              data-testid="edit-link"
              href="/users/undefined/edit"
            >
              <svg
                class="inline pb-0.5"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 20h9"
                />
                <path
                  d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
                />
              </svg>
              <span
                class="text-sm px-1"
              >
                Edit
              </span>
            </a>
          </div>
          <div
            class="bg-white border-2 border-dial-gray lg:mr-6 shadow-lg"
          >
            <div
              class="flex flex-col h-80 p-4"
            >
              <div
                class="m-auto align-middle w-48"
              >
                <img
                  alt="Logo for: Some User"
                  class="w-24 m-auto"
                  src="/icons/user.svg"
                />
              </div>
              <div
                class="text-sm text-center text-dial-gray-dark"
              >
                Email
                 : 
                some_user@some-website.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full lg:w-2/3 xl:w-3/4 flex flex-col"
        >
          <div
            class="hidden lg:block"
          >
            <div
              class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
            >
              <a
                class="inline text-dial-sapphire h5"
                href="/"
              >
                Home
              </a>
              <div
                class="inline h5"
              >
                 &gt; 
                <a
                  class="text-dial-gray-dark"
                  href="/"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                User Profile
                :
              </div>
              <div>
                Some User
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                Roles
                :
              </div>
              <div>
                USER
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Toastify"
    />
    <div
      class="flex flex-col lg:flex-row justify-between pb-8"
    >
      <div
        class="relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4"
      >
        <div
          class="pb-4 flex"
        >
          <a
            class="cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
            data-testid="edit-link"
            href="/users/undefined/edit"
          >
            <svg
              class="inline pb-0.5"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 20h9"
              />
              <path
                d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
              />
            </svg>
            <span
              class="text-sm px-1"
            >
              Edit
            </span>
          </a>
        </div>
        <div
          class="bg-white border-2 border-dial-gray lg:mr-6 shadow-lg"
        >
          <div
            class="flex flex-col h-80 p-4"
          >
            <div
              class="m-auto align-middle w-48"
            >
              <img
                alt="Logo for: Some User"
                class="w-24 m-auto"
                src="/icons/user.svg"
              />
            </div>
            <div
              class="text-sm text-center text-dial-gray-dark"
            >
              Email
               : 
              some_user@some-website.com
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full lg:w-2/3 xl:w-3/4 flex flex-col"
      >
        <div
          class="hidden lg:block"
        >
          <div
            class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
          >
            <a
              class="inline text-dial-sapphire h5"
              href="/"
            >
              Home
            </a>
            <div
              class="inline h5"
            >
               &gt; 
              <a
                class="text-dial-gray-dark"
                href="/"
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-3"
        >
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              User Profile
              :
            </div>
            <div>
              Some User
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              Roles
              :
            </div>
            <div>
              USER
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Unit test for the UserDetail component should render element of the user owning product. 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Toastify"
      />
      <div
        class="flex flex-col lg:flex-row justify-between pb-8"
      >
        <div
          class="relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4"
        >
          <div
            class="pb-4 flex"
          >
            <a
              class="cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
              data-testid="edit-link"
              href="/users/undefined/edit"
            >
              <svg
                class="inline pb-0.5"
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 20h9"
                />
                <path
                  d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
                />
              </svg>
              <span
                class="text-sm px-1"
              >
                Edit
              </span>
            </a>
          </div>
          <div
            class="bg-white border-2 border-dial-gray lg:mr-6 shadow-lg"
          >
            <div
              class="flex flex-col h-80 p-4"
            >
              <div
                class="m-auto align-middle w-48"
              >
                <img
                  alt="Logo for: Some User"
                  class="w-24 m-auto"
                  src="/icons/user.svg"
                />
              </div>
              <div
                class="text-sm text-center text-dial-gray-dark"
              >
                Email
                 : 
                some_user@some-website.com
              </div>
            </div>
          </div>
        </div>
        <div
          class="w-full lg:w-2/3 xl:w-3/4 flex flex-col"
        >
          <div
            class="hidden lg:block"
          >
            <div
              class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
            >
              <a
                class="inline text-dial-sapphire h5"
                href="/"
              >
                Home
              </a>
              <div
                class="inline h5"
              >
                 &gt; 
                <a
                  class="text-dial-gray-dark"
                  href="/"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-3"
          >
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                User Profile
                :
              </div>
              <div>
                Some User
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                Roles
                :
              </div>
              <div>
                ORG_USER, PRODUCT_USER
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                Products
                :
              </div>
              <div>
                Some Product
              </div>
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="font-semibold"
              >
                Organization
                :
              </div>
              <div>
                Some Organization
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Toastify"
    />
    <div
      class="flex flex-col lg:flex-row justify-between pb-8"
    >
      <div
        class="relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4"
      >
        <div
          class="pb-4 flex"
        >
          <a
            class="cursor-pointer bg-dial-iris-blue px-2 py-0.5 rounded text-white"
            data-testid="edit-link"
            href="/users/undefined/edit"
          >
            <svg
              class="inline pb-0.5"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 20h9"
              />
              <path
                d="M16.5 3.5a2.121 2.121 0 0 1 3 3L7 19l-4 1 1-4L16.5 3.5z"
              />
            </svg>
            <span
              class="text-sm px-1"
            >
              Edit
            </span>
          </a>
        </div>
        <div
          class="bg-white border-2 border-dial-gray lg:mr-6 shadow-lg"
        >
          <div
            class="flex flex-col h-80 p-4"
          >
            <div
              class="m-auto align-middle w-48"
            >
              <img
                alt="Logo for: Some User"
                class="w-24 m-auto"
                src="/icons/user.svg"
              />
            </div>
            <div
              class="text-sm text-center text-dial-gray-dark"
            >
              Email
               : 
              some_user@some-website.com
            </div>
          </div>
        </div>
      </div>
      <div
        class="w-full lg:w-2/3 xl:w-3/4 flex flex-col"
      >
        <div
          class="hidden lg:block"
        >
          <div
            class="bg-white pb-3 lg:py-4 whitespace-nowrap text-ellipsis overflow-hidden"
          >
            <a
              class="inline text-dial-sapphire h5"
              href="/"
            >
              Home
            </a>
            <div
              class="inline h5"
            >
               &gt; 
              <a
                class="text-dial-gray-dark"
                href="/"
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-col gap-3"
        >
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              User Profile
              :
            </div>
            <div>
              Some User
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              Roles
              :
            </div>
            <div>
              ORG_USER, PRODUCT_USER
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              Products
              :
            </div>
            <div>
              Some Product
            </div>
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="font-semibold"
            >
              Organization
              :
            </div>
            <div>
              Some Organization
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
